src/overlays/actors/ovl_En_Fsn/z_en_fsn.h

```diff
@@ -2,12 +2,101 @@
 #define Z_EN_FSN_H
 
 #include <global.h>
+#include "overlays/actors/ovl_En_GirlA/z_en_girla.h"
+
+#define ENFSN_IS_BACKROOM(thisx) ((thisx)->params & 1)
+#define ENFSN_IS_SHOP(thisx) (!((thisx)->params & 1))
+
+#define ENFSN_SHOULD_HAGGLE(this) ((this)->flags & 0x80)
+#define ENFSN_IS_ANGRY(this) ((this)->flags & 0x100)
+#define ENFSN_SHOULD_CALM_DOWN(this) ((this)->flags & 0x200)
+#define ENFSN_SHOULD_OFFER_FINAL_PRICE(this) ((this)->flags & 0x40)
+
+#define ENFSN_SET_SHOULD_HAGGLE(this) ((this)->flags |= 0x80)
+#define ENFSN_SET_IS_ANGRY(this) ((this)->flags |= 0x100)
+#define ENFSN_SET_SHOULD_CALM_DOWN(this) ((this)->flags |= 0x200)
+#define ENFSN_SET_SHOULD_OFFER_FINAL_PRICE(this) ((this)->flags |= 0x40)
+
+#define ENFSN_UNSET_SHOULD_HAGGLE(this) ((this)->flags &= ~0x80)
+#define ENFSN_UNSET_IS_ANGRY(this) ((this)->flags &= ~0x100)
+#define ENFSN_UNSET_SHOULD_CALM_DOWN(this) ((this)->flags &= ~0x200)
+#define ENFSN_UNSET_SHOULD_OFFER_FINAL_PRICE(this) ((this)->flags &= ~0x40)
 
 struct EnFsn;
 
+typedef void (*EnFsnActionFunc)(struct EnFsn*, GlobalContext*);
+
+typedef struct {
+    /* 0x00 */ u32 stickColorR;
+    /* 0x04 */ u32 stickColorG;
+    /* 0x08 */ u32 stickColorB;
+    /* 0x0C */ u32 stickColorA;
+    /* 0x10 */ f32 stickTexX;
+    /* 0x14 */ f32 stickTexY;
+    /* 0x18 */ u32 arrowColorR;
+    /* 0x1C */ u32 arrowColorG;
+    /* 0x20 */ u32 arrowColorB;
+    /* 0x24 */ u32 arrowColorA;
+    /* 0x28 */ f32 arrowTexX;
+    /* 0x2C */ f32 arrowTexY;
+    /* 0x30 */ f32 texZ; // Used for both stickTexZ and arrowTexZ
+    /* 0x34 */ s32 isEnabled;
+} StickDirectionPrompt; // size = 0x38
```

Comment:
I've been working on the shopkeeper actors (EnOssan, EnSob1, EnTrt, and EnFsn). They all share this common struct. What does everyone think is the best approach?

1. Duplicate the struct definition in each header
2. Create a new header (e.g. shopkeeper.h) with common struct definitions/macros
3. Store common struct defintions/macros in `en_girla.h` (shop items) since all shopkeepers have to include this header anyways for their items.

Reply:
I'm partial to option 3.